Test/CoreSDK.Test/Net46/Extensibility/Implementation/RichPayloadEventSourceTest.cs

```diff
@@ -95,61 +106,163 @@ public void RichPayloadEventSourcePageViewSentTest()
-                object[] tags = actualEvent.Payload[1] as object[];
-                foreach (object tagObject in tags)
-                {
-                    Dictionary<string, object> tag = (Dictionary<string, object>)tagObject;
-                    Assert.IsNotNull(tag);
-                    string key = (string)tag["Key"];
-                    object value = tag["Value"];
-                    if (!string.IsNullOrWhiteSpace(key))
+                    Assert.IsNotNull(actualEvent);
+                    Assert.AreEqual(client.InstrumentationKey, actualEvent.Payload[0]);
+
+                    int keysFound = 0;
+                    object[] tags = actualEvent.Payload[1] as object[];
+                    foreach (object tagObject in tags)
                     {
-                        if (key == "ai.user.id")
+                        Dictionary<string, object> tag = (Dictionary<string, object>)tagObject;
+                        Assert.IsNotNull(tag);
+                        string key = (string)tag["Key"];
+                        object value = tag["Value"];
+                        if (!string.IsNullOrWhiteSpace(key))
                         {
-                            Assert.AreEqual("testUserId", value);
-                            ++keysFound;
+                            if (key == "ai.user.id")
+                            {
+                                Assert.AreEqual("testUserId", value);
+                                ++keysFound;
+                            }
+                            else if (key == "ai.operation.id")
+                            {
+                                Assert.AreEqual(item.Context.Operation.Id, value);
+                                ++keysFound;
+                            }
                         }
-                        else if (key == "ai.operation.id")
+                    }
+
+                    Assert.AreEqual(2, keysFound);
+                    Assert.IsNotNull(actualEvent.Payload[2]);
+
+                    var expectedProperties = dataType.GetProperties().AsEnumerable();
+                    var actualPropertiesPayload = (IDictionary<string, object>)actualEvent.Payload[2];
+                    VerifyEventPayload(expectedProperties, actualPropertiesPayload);
+                }
+            }
+            else
+            {
+                // 4.5 doesn't have RichPayload events
+                Assert.IsNull(RichPayloadEventSource.Log.EventSourceInternal);
+            }
+        }
+
+        private static void VerifyEventPayload(IEnumerable<PropertyInfo> expectedProperties, IDictionary<string, object> actualEventPayload)
+        {
+            var actualProperties = actualEventPayload.Keys.Select(k => new { Key = k, Value = actualEventPayload[k] });
+
+            Assert.IsTrue(expectedProperties.Count() == actualProperties.Count());
+            var e1 = expectedProperties.GetEnumerator();
+            var e2 = actualProperties.GetEnumerator();
```

Comment:
Thanks. Will change the variable name. For the var, I don't think "var" in this case impacts understanding the code as long as the variable name provide the essential information. But if you insist it's a coding style in AI Lib I need to follow, I will change it.